### PR TITLE
fix: `to_any_value` should supports all LiteralValue type

### LINKED ISSUE #15379: `pl.repeat()` panicks if input doesn't match data type
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
>>> import polars as pl
>>> pl.repeat([1, 2], n=2)
thread 'python3' panicked at py-polars/src/functions/lazy.rs:454:36:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/etienne/.local/lib/python3.10/site-packages/polars/functions/repeat.py", line 143, in repeat
    expr = wrap_expr(plr.repeat(value, n._pyexpr, dtype))
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```

### Log output

_No response_

### Issue description

`pl.repeat()` panicks when input length is greater than 1.

### Expected behavior

Not panick but throw an error instead.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.17
Index type:           UInt32
Platform:             Linux-6.5.0-21-generic-x86_64-with-glibc2.35
Python:               3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                1.21.5
openpyxl:             <not installed>
pandas:               <not installed>
pyarrow:              <not installed>
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by reswqa**
This fixes #15379.

### CODE REVIEW

**Review by stinodego — CHANGES_REQUESTED** [review]
I think we should instead implement `to_any_value` for the missing types. Should be easy enough - Series and Range become a List type AnyValue, and Binary becomes a Binary AnyValue.

File: crates/polars-plan/src/logical_plan/lit.rs [hunk 1]
```diff
@@ -116,7 +117,43 @@ impl LiteralValue {
...
+                        if *low < 0 || *high > u32::MAX as i64 {
+                            return None;
+                        }
+                        let low = *low as u32;
+                        let high = *high as u32;
+                        let ca: NoNull<UInt32Chunked> = (low..high).collect();
+                        ca.into_inner().into_series()
+                    },
+                    _ => return None,
+                };
```
Comment (stinodego) [inline crates/polars-plan/src/logical_plan/lit.rs:153]: I think you can use just `new_int_range` here, right?

**Review by stinodego — APPROVED** [review]
Looks good! Nice that we can use `repeat` on list inputs now 👌

**Review by stinodego — APPROVED** [review]
Looks good! Nice that we can use `repeat` on list inputs now 👌

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**